src/state/quick-open-mixin.ts

```diff
@@ -0,0 +1,56 @@
+import { showDialog } from "../dialogs/make-dialog-manager";
+import type { Constructor } from "../types";
+import type { HassBaseEl } from "./hass-base-mixin";
+import type { QuickOpenDialogParams } from "../dialogs/quick-open/ha-quick-open-dialog";
+import type { PropertyValues } from "lit-element";
+import type { HASSDomEvent } from "../common/dom/fire_event";
+
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "hass-quick-open": QuickOpenDialogParams;
+    "hass-service-called": HassServiceCalledParams;
+  }
+}
+
+export interface HassServiceCalledParams {
+  domain: string;
+  service: string;
+  serviceData?: {};
+}
+
+let quickOpenImportPromise;
+const importQuickOpen = () => {
+  if (!quickOpenImportPromise) {
+    quickOpenImportPromise = import(
+      /* webpackChunkName: "quick-open-dialog" */ "../dialogs/quick-open/ha-quick-open-dialog"
+    );
+  }
+  return quickOpenImportPromise;
+};
+
+export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
+  class extends superClass {
+    protected firstUpdated(changedProps: PropertyValues) {
+      super.firstUpdated(changedProps);
+      this.addEventListener("hass-quick-open", (ev) =>
+        this._handleQuickOpen(ev)
+      );
+
+      // Load it once we are having the initial rendering done.
+      importQuickOpen();
```

Comment:
I think we can just lazy load this?

Reply:
As in, move this import call into the keyboard listener block?

Reply:
No, check how we do other dialogs

Reply:
This mixin was modeled after `more-info-mixin` which has the same call to a lazy import promise. disconnect-toast-mixin does the same thing in `firstUpdated` but more directly.

Which mixin is more appropriate to emulate here?

(Also, take a look at what I just pushed up. I've added the keyboard listener to the mixin and put the import in the listener block. I believe this should only import the dialog when the user types the hotkey)

Reply:
Just the many `show-dialog` functions like: <https://github.com/home-assistant/frontend/blob/dev/src/panels/config/users/show-dialog-add-user.ts>

Reply:
@bramkragten I made some changes that moved us toward the "show-dialog" pattern. Could you take a look and let me know if it needs more adjustment?

Reply:
Looks good!